python/pyarrow/fs.py

```diff
@@ -117,6 +116,9 @@ def get_type_name(self):
             protocol = protocol[0]
         return "fsspec+{0}".format(protocol)
 
+    def normalize_path(self, path):
+        return path
```

Comment:
@jorisvandenbossche This seems to work for now. I'm not sure if `fsspec` normalizes paths internally (especially on Windows).

Reply:
From a quick looks, it seems that their LocalFileSystem should handle incoming "windows" (non-normalized) paths (see eg https://github.com/intake/filesystem_spec/pull/65/files, where a bunch of `path = make_path_posix(path)` were added to the LocalFileSystem methods). So I think this should be fine.

Only, do we rely on this method on the C++ side to actually return a normalized path? 
Eg in the C++ dataset discovery code, that might be relying on being able to split the normalized path on `/` ?

Reply:
When calling `GetFileInfo(selector)`, we need the return values to be a lexical child of the selector base_dir.

Reply:
So if `GetFileInfo` normalizes, so should `NormalizePath`. Otherwise, not necessary.

Reply:
That said, we can also special-case `fsspec`'s LocalFileSystem to use our own LocalFileSystem instead... do we already do that?

Reply:
We indeed already do that:

https://github.com/apache/arrow/blob/90d1ab73a132faaaa0a205a8640d78bf1c60005f/python/pyarrow/fs.py#L77-L79

at least in cases where the `_ensure_filesystem` is used to process the user specified filesystem before passing it to the underlying functions (but which should in principle be done by all user facing functions).

So the actual fsspec's LocalFileSystem is only used for testing, so I think this is certainly fine then (since other filesystems should not have this problem of windows paths)?

Reply:
Let's assume it's fine, then.